src/vm/appdomain.cpp

```diff
@@ -3693,7 +3693,8 @@ void SystemDomain::ExecuteMainMethod(HMODULE hMod, __in_opt LPWSTR path /*=NULL*
         pDomain->GetMulticoreJitManager().AutoStartProfile(pDomain);
 #endif
 
-        pDomain->m_pRootAssembly->ExecuteMainMethod(NULL);
+        static const BOOL WaitForOtherThreads = TRUE;
+        pDomain->m_pRootAssembly->ExecuteMainMethod(NULL, !WaitForOtherThreads);
```

Comment:
The typical way to write this in the CoreCLR codebase is:

`ExecuteMainMethod(NULL, FALSE /* waitForOtherThreads */);`